ui-showdetails/src/main/java/app/tivi/showdetails/details/ShowDetailsViewModel.kt

```diff
@@ -72,35 +77,36 @@ internal class ShowDetailsViewModel @Inject constructor(
     private val loadingState = ObservableLoadingCounter()
     private val uiMessageManager = UiMessageManager()
 
-    val state = combine(
-        observeShowFollowStatus.flow,
-        observeShowDetails.flow,
-        observeShowImages.flow,
-        loadingState.observable,
-        observeRelatedShows.flow,
-        observeNextEpisodeToWatch.flow,
-        observeShowSeasons.flow,
-        observeShowViewStats.flow,
-        uiMessageManager.message,
-    ) { isFollowed, show, showImages, refreshing, relatedShows, nextEpisode, seasons, stats,
-        message ->
+    // Need to add a MonotonicFrameClock
+    // See: https://github.com/cashapp/molecule/#frame-clock
+    private val moleculeScope = CoroutineScope(viewModelScope.coroutineContext + AndroidUiDispatcher.Main)
```

Comment:
Since this is a new `CoroutineScope` instance that is unmanaged, doesn't it have to be explicitly cancelled in the `onCleared()`? 🤔

Reply:
It's not unmanaged, it's reusing the `viewModelScope` (specifically it's `Job`)

Reply:
Sweet! I learned something new about `CoroutineScopes` 😃 .

I even tested it with:
https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS42LjIxIiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoiaW1wb3J0IGtvdGxpbnguY29yb3V0aW5lcy4qXG5cbmZ1biBtYWluKCkgPSBydW5CbG9ja2luZyB7XG4gICAgdmFsIHNjb3BlMSA9IENvcm91dGluZVNjb3BlKEpvYigpKVxuICAgIHZhbCBzY29wZTIgPSBzY29wZTEgKyBEaXNwYXRjaGVycy5EZWZhdWx0XG4gICAgc2NvcGUyLmxhdW5jaCB7IGRlbGF5KDEwMDBMKSB9XG4gICAgXG4gICAgcHJpbnRsbihcInNjb3BlMi5pc0FjdGl2ZSA9ICR7c2NvcGUyLmlzQWN0aXZlfVwiKVxuICAgIHByaW50bG4oXCJjYW5jZWxsaW5nIHNjb3BlMVwiKVxuICAgIHNjb3BlMS5jYW5jZWwoKVxuICAgIHByaW50bG4oXCJzY29wZTEgY2FuY2VsbGVkXCIpXG4gICAgcHJpbnRsbihcInNjb3BlMi5pc0FjdGl2ZSA9ICR7c2NvcGUyLmlzQWN0aXZlfVwiKVxufSJ9

Result:
```
scope2.isActive = true
cancelling scope1
scope1 cancelled
scope2.isActive = false
```